training/src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -25,11 +28,31 @@
     @Autowired
     private BookRepository bookRepository;
 
+    @Autowired
+    private OpenLibraryService openLibraryService;
+
     @GetMapping
     public Iterable findAll() {
         return bookRepository.findAll();
     }
 
+    @GetMapping("/isbn/{isbn}")
+    public Book bookInfoByIsbn(@PathVariable String isbn)
+        throws BookNotFoundException, ParseException, JsonProcessingException {
+
+        Book bookInRepository = bookRepository.findFirstByIsbn(isbn);
```

Comment:
Remember that you should first check if the book is present on your DB. If not, you should go check the API.